modules/bootstrapped/test/src/smithy4s/builder/EndpointBuilderSpec.scala

```diff
@@ -0,0 +1,58 @@
+package smithy4s
+
+import munit._
+
+class EndpointBuilderSpec extends FunSuite {
+
+  val endpoint = smithy4s.example.WeatherGen.endpoints.apply(2)
+
+  val builder = smithy4s.Endpoint.Builder.fromEndpoint(endpoint)
+
+  test(
+    "can replace the following values (Id and Hints) using withId and withHints"
+  ) {
+
+    val newEndpoint = builder
+      .withId(ShapeId.apply("smithy4s.example", "endpoint"))
+      .withHints(Hints(Hints.Binding.DynamicBinding.apply(ShapeId.apply("smithy.api", "documentation"), Document.fromString("this is a endpoint"))))
+      .withErrorable(None)
+      .build
+
+    assertEquals(newEndpoint.id, ShapeId.apply("smithy4s.example", "endpoint"))
+    assertEquals(newEndpoint.hints, Hints(Hints.Binding.DynamicBinding.apply(ShapeId.apply("smithy.api", "documentation"), Document.fromString("this is a endpoint"))))
+    assertEquals(newEndpoint.errorable, None)
+
+  }
+
+
+  test(
+    "can modify the following values (Id, Hints, Input, Output, Errorable) using mapId, mapHints, mapInput, mapOutput, mapErrorable"
+  ) {
+
+
+    val newEndpoint = builder
+      .mapId(shapeId => ShapeId.apply(shapeId.namespace, "getCityEndpoint"))
+      .mapHints { hints =>
+        hints.++(
+          Hints(
+            Hints.Binding.DynamicBinding.apply(ShapeId.apply("smithy.api", "documentation"), Document.fromString("new endpoint")))
+        )
+      }
+      .mapInput(s => s.withId(ShapeId.apply("smithy4s.example", "inputSchema")))
+      .mapOutput(s => s.withId(ShapeId.apply("smithy4s.example", "outputSchema")))
+      .mapErrorable(_ => None)
```

Comment:
**thought:** this could have some interesting implicitations if someone does that on an endpoint that has errors: I wonder what happens if they use the "surfaced errors" way of writing services and raise an error that way...

Reply:
I think it's worth a test in here, at least.

Reply:
just checked, the behavior is:

- if you remove an errorable and the service raises an error, it behaves just like on any other Throwable
- adding errors to an existing errorable, or creating custom Alts in the errorable otherwise, seems impossible (legally) because `E` isn't known: so you can't biject an arbitrary value into it

Reply:
https://github.com/disneystreaming/smithy4s/pull/1097/commits/f31a91264de502e1a1e2b721bcd61c746f60b541